### PR TITLE
Qr security event

### PR DESCRIPTION
**Pull request by shainw**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: Detections/SecurityEvent/RDP_Nesting.yaml [hunk 1]
```diff
@@ -21,35 +21,38 @@ query: |
...
-  // use left anti to exclude anything from the previous 7 days where the Account and IP has connected to more than 5 computers.
+  // use left anti to exclude anything from the previous 7 days where the Account and IP has connected 5 or more computers.
   | join kind=leftanti (
   SecurityEvent
   | where TimeGenerated >= ago(starttime) and TimeGenerated < ago(endtime) 
   | where EventID == 4624 and LogonType == 10
-  | summarize makeset(Computer), ComputerCount = dcount(Computer) by Account, IpAddress
-  // Connection count to computer by same account and IP to determine rarity
-  | where ComputerCount > threshold
+  | summarize makeset(Computer), ComputerCount = dcount(Computer) by bin(TimeGenerated, 1d), Account = tolower(Account), IpAddress
```
Comment (petebryan) [inline Detections/SecurityEvent/RDP_Nesting.yaml]: make_set has replaced makeset so maybe consider updating this. (Makeset still works)

File: Detections/SecurityEvent/RDP_MultipleConnectionsFromSingleSystem.yaml [hunk 2]
```diff
@@ -20,27 +21,20 @@ query: |
...
-  | where EventID == 4624 and LogonType == 10
-  | summarize ComputerCount = dcount(Computer) by bin(TimeGenerated, 1d), Account, IpAddress
-  // Computer count by same Account divided by days to determine average count
-  | extend AvgComputerCount = ComputerCount/avgDays;
+  let threshold = 2.0;
   SecurityEvent
   | where TimeGenerated >= ago(endtime) 
   | where EventID == 4624 and LogonType == 10
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ComputerCount = dcount(Computer), ComputerSet = makeset(Computer), ProcessSet = makeset(ProcessName)  
+  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), ComputerCountToday = dcount(Computer), ComputerSet = makeset(Computer), ProcessSet = makeset(ProcessName)
```
Comment (petebryan) [inline Detections/SecurityEvent/RDP_MultipleConnectionsFromSingleSystem.yaml]: Consider updating makeset with make_set

File: Detections/SecurityEvent/PotentialKerberoast.yaml [hunk 3]
```diff
@@ -28,35 +29,35 @@ query: |
...
+  | parse EventData with * 'ServiceName">' ServiceName "<" *
   | where ServiceName !contains "$" and ServiceName !contains "krbtgt" 
-  | where TargetUserName !contains ServiceName;
+  | parse EventData with * 'TargetUserName">' TargetUserName "<" *
+  | where TargetUserName !contains "$@" and TargetUserName !contains ServiceName
+  | parse EventData with * 'IpAddress">::ffff:' ClientIPAddress "<" *;
   let Kerbevent23h = Kerbevent
   | where TimeGenerated >= ago(starttime) and TimeGenerated < ago(endtime)
-  | summarize ServiceNameCountPrev23h = dcount(ServiceName), ServiceNameSet23h = makeset(ServiceName) by Computer, TargetUserName, Cl_IPAddress, TicketOptions, TicketEncryptionType, Status
+  | summarize ServiceNameCountPrev23h = dcount(ServiceName), ServiceNameSet23h = makeset(ServiceName)
```
Comment (petebryan) [inline Detections/SecurityEvent/PotentialKerberoast.yaml]: Consider updating makeset to make_set

File: Detections/SecurityEvent/PotentialKerberoast.yaml [hunk 4]
```diff
@@ -28,35 +29,35 @@ query: |
...
   | where TimeGenerated >= ago(starttime) and TimeGenerated < ago(endtime)
-  | summarize ServiceNameCountPrev23h = dcount(ServiceName), ServiceNameSet23h = makeset(ServiceName) by Computer, TargetUserName, Cl_IPAddress, TicketOptions, TicketEncryptionType, Status
+  | summarize ServiceNameCountPrev23h = dcount(ServiceName), ServiceNameSet23h = makeset(ServiceName) 
+  by Computer, TargetUserName, ClientIPAddress, TicketOptions, TicketEncryptionType, Status
   | where ServiceNameCountPrev23h < prev23hThreshold;
   let Kerbevent1h = 
   Kerbevent
   | where TimeGenerated >= ago(endtime)
-  | summarize min(TimeGenerated), max(TimeGenerated), ServiceNameCountPrev1h = dcount(ServiceName), ServiceNameSet1h = makeset(ServiceName) by Computer, TargetUserName, Cl_IPAddress, TicketOptions, TicketEncryptionType, Status;
+  | summarize min(TimeGenerated), max(TimeGenerated), ServiceNameCountPrev1h = dcount(ServiceName), ServiceNameSet1h = makeset(ServiceName)
```
Comment (petebryan) [inline Detections/SecurityEvent/PotentialKerberoast.yaml]: consider updating makeset to make_set

File: Detections/SecurityEvent/ExcessiveLogonFailures.yaml [hunk 5]
```diff
@@ -17,20 +17,43 @@ relevantTechniques:
...
+  SubStatus == '0xc0000072', 'Account is disabled',
+  SubStatus == '0xc0000133', 'Clocks between DC and other computer too far out of sync',
+  SubStatus == '0xc000015b', 'The user has not been granted the requested logon right at this machine',
+  SubStatus == '0xc0000193', 'Account has expirated',
+  SubStatus == '0xc0000224', 'User is required to change password at next logon',
+  SubStatus == '0xc0000234', 'Account is currently locked out',
+  strcat('Unknown reason substatus: ', SubStatus))
+  | extend WorkstationName = iff(WorkstationName == "-" or isempty(WorkstationName), Computer , WorkstationName) 
+  | project StartTimeUtc, EndTimeUtc, EventID, Account, LogonTypeName, SubStatus, Reason, AccountType, Computer, WorkstationName, IpAddress, CountToday, CountPrev7day, Avg7Day = CountPrev7day/7
+  | summarize StartTimeUtc = min(StartTimeUtc), EndTimeUtc = max(EndTimeUtc), Computer = makelist(Computer), IpAddressList = makelist(IpAddress), sum(CountToday), sum(CountPrev7day), avg(Avg7Day)
```
Comment (petebryan) [inline Detections/SecurityEvent/ExcessiveLogonFailures.yaml]: Consider updating makelist to make_list

File: Detections/SecurityEvent/gte_6_FailedLogons_10m.yaml [hunk 6]
```diff
@@ -18,11 +18,14 @@ relevantTechniques:
 query: |
 
   let timeframe = 10m;
+  let threshold = 20;
   SecurityEvent 
-  | where TimeGenerated > ago(2*timeframe) 
+  | where TimeGenerated >= ago(timeframe)
   | where EventID == "4625"
```
Comment (petebryan) [inline Detections/SecurityEvent/gte_6_FailedLogons_10m.yaml]: In other queries you have dropped the " around the EventID number

File: Detections/SecurityEvent/RDP_Nesting.yaml (hunk 1 above)
Comment (shainw) [reply Detections/SecurityEvent/RDP_Nesting.yaml]: yeah, Ashwin and I discussed this.  It hasn't replaced it in the Log Analytics framework, while it works, it gets redlined and it won't complete intellisense.  Until LA adopts it, I don't think we should be using it.

File: Detections/SecurityEvent/RDP_MultipleConnectionsFromSingleSystem.yaml (hunk 2 above)
Comment (shainw) [reply Detections/SecurityEvent/RDP_MultipleConnectionsFromSingleSystem.yaml]: Same as above.

File: Detections/SecurityEvent/PotentialKerberoast.yaml (hunk 3 above)
Comment (shainw) [reply Detections/SecurityEvent/PotentialKerberoast.yaml]: same as above

File: Detections/SecurityEvent/PotentialKerberoast.yaml (hunk 4 above)
Comment (shainw) [reply Detections/SecurityEvent/PotentialKerberoast.yaml]: same as above

File: Detections/SecurityEvent/ExcessiveLogonFailures.yaml (hunk 5 above)
Comment (shainw) [reply Detections/SecurityEvent/ExcessiveLogonFailures.yaml]: Same as above

File: Detections/SecurityEvent/gte_6_FailedLogons_10m.yaml (hunk 6 above)
Comment (shainw) [reply Detections/SecurityEvent/gte_6_FailedLogons_10m.yaml]: Yes, thanks.

**Pull request merged**